src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,555 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    internal sealed class ControlFlowGraphBuilder : OperationCloner
+    {
+        private readonly BasicBlock _entry = new BasicBlock(BasicBlockKind.Entry);
```

Comment:
`_entry` and `_exit` could be static?

Reply:
> _entry and _exit could be static?

They are part of each graph and are linked to other basic blocks in it.

---
In reply to: [162144163](https://github.com/dotnet/roslyn/pull/24263#discussion_r162144163) [](ancestors = 162144163)

Reply:
Then I don't think they should be fields. They can be created where they are used (which is one location each).

Reply:
We are not finished yet, new uses are coming. For example, when we will visit return statement we will link the current block to the _exit. 

---
In reply to: [162262981](https://github.com/dotnet/roslyn/pull/24263#discussion_r162262981) [](ancestors = 162262981)